sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -354,8 +364,8 @@ public Double getDouble(String propertyName) {
                 }
             } else if (JsonSerializable.class.isAssignableFrom(c)) {
                 return (T) ModelBridgeInternal.instantiateJsonSerializable((ObjectNode) jsonObj, c);
-            } else if (JsonSerializableWrapper.class.isAssignableFrom(c)) {
-                return (T) ModelBridgeInternal.instantiateJsonSerializableWrapper((ObjectNode) jsonObj, c);
+            } else if (containsJsonSerializable(c)) {
+                return ModelBridgeInternal.instantiateByJsonString(Utils.toJson(Utils.getSimpleObjectMapper(), (ObjectNode) jsonObj), c);
```

Comment:
doesn't this result in double json serialization/deserialization?

can we pass ObjectNode instead? otherwise this will serialize to string and the corresponding class will deserialize string back to ObjectNode.

Reply:
Changed to use ObjectNode related constuctor.